exopy/measurement/plugin.py

```diff
@@ -42,6 +42,29 @@ def _workspace():
     return MeasurementSpace
 
 
+class TaskRuntimeContext():
+    """A context manager used to give temporary access to a task runtime
+    (e.g. driver) to a task.
+
+    """
+    def __init__(self, dependencies, task):
+        self.dependencies = dependencies
+        self.task = task
+
+    def __enter__(self):
+        res, msg, errors = self.dependencies.collect_task_runtimes(self.task)
+        if res:
+            self.task.root.run_time = self.dependencies.get_runtime_dependencies('main')
+        else:
+            logger.error(msg)
+            logger.error(errors)
```

Comment:
Having a way to know inside the context manager if the collection succeeded would be nice. That would allow a single point of check rather than having to catch arbitrary exception whose root cause may be hard to track down.

Reply:
I think the logs go a good job of helping track down the root cause but now `__enter__` also returns the errors so the caller can decide what to do with them.